qiskit/synthesis/su4/utils.py

```diff
@@ -0,0 +1,88 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2023
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+# pylint: disable=invalid-name
+
+"""Utilities for SU(4) synthesis: Weyl chamber, polytopes, distances, etc."""
+
+import numpy as np
+
+
+_EPS = 1e-12
+
+
+def find_min_point(P):
+    """Find the closest point on convex polytope P to the origin (in Euclidean distance).
+
+    P is given as a list of points forming its vertices.
+
+    Function from
+    https://scipy-cookbook.readthedocs.io/items/Finding_Convex_Hull_Minimum_Point.html
```

Comment:
It's not clear to me what the licence of this code is.  The best I can find is in the repo of the docs website, which has a stipulation that source-code reproductions must include their licence header and binary distributions must include it in their documentation.  The licence comes from the times when Scipy was part of Enthought.

I think we need to be more careful about just copy-pasting code like this - it's not clear to me if we're actually safe to use this.